### PR TITLE
Bug fixes for Alibaba Cloud and minor improvements

### PR DESCRIPTION
**Pull request by marcplouhinec**
Fix several bugs when Benchmarking Alibaba Cloud services.
Moreover, add a new optional flag in the ping benchmark in order to additionally execute ping tests by using public IPs between virtual machines.

### CODE REVIEW

**Comment by marcplouhinec** [comment]
I signed it!

**Comment by gareth-ferneyhough** [comment]
Thanks for the PR! Someone will take a look when we have a free moment.

File: perfkitbenchmarker/linux_packages/fio.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ def _Install(vm):
   """Installs the fio package on the VM."""
   for p in ['build_tools', 'python', 'pip']:
     vm.Install(p)
+  vm.InstallPackages('python-dev')
```
Comment (gareth-ferneyhough) [inline perfkitbenchmarker/linux_packages/fio.py]: Can you explain why do you need this?

Comment (gareth-ferneyhough) [reply perfkitbenchmarker/linux_packages/fio.py]: Fair enough. Thanks for the PR.

**Pull request merged**